test/yargs-parser.js

```diff
@@ -2200,6 +2200,26 @@ describe('yargs-parser', function () {
 
         parsed['x'].should.equal('b')
       })
+      it('array[string]: keeps only last argument', function () {
+        var parsed = parser('-x a -x b', {
+          configuration: {
+            array: [{ key: 'x', string: true }],
+            'duplicate-arguments-array': false
+          }
+        })
+
+        parsed['x'].should.equal('b')
```

Comment:
this behavior seems wrong to me, if we've explicitly indicated that `x` should be an array, the value should be `['b']` not `b` -- I also wonder if maybe `array` should take precedence over `duplicate-arguments-array`, and it should be `['a', 'b']` (I think this would be least shocking).

Reply:
Yes, it's wrong, but I didn't open this coffin. The option type `string[]` behavior remains unchanged.

I tried to copy the option type `string[]` behavior to the option type `number[]`, to get consistent results.

Reply:
done